### PR TITLE
[bug][auth]Show grant causes role errors in the memory. #24783

### LINKED ISSUE #24783: [Bug]Show grant causes role errors in the memory.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

Show grant causes role errors in the memory.

### What You Expected?

role priv correct after show grant

### How to Reproduce?
Test in doris2.0.1 and after fix revoke cause fe crash issue

drop role role5;
drop role role6;
create role role5;
create role role6;
grant Select_priv ON *.*.* TO role 'role5';
grant Drop_priv ON *.*.* TO role 'role6';
grant 'role5' to `zltest4`;
grant 'role6' to `zltest4`;
show all grants;
revoke 'role6' from `zltest4`;
show all grants;

image

image

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fsilent**
## Proposed changes

Issue Number: close #24783

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by fsilent** [comment]
run buildall

**Comment by fsilent** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/PrivEntry.java [hunk 1]
```diff
@@ -187,4 +188,29 @@ protected static int compareAssist(String... strings) {
         }
         return 0;
     }
+
+    protected PrivEntry copy() throws AnalysisException, PatternMatcherException {
```
Comment (zddr) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/PrivEntry.java:192]: change this method to `abstract`,then write the `copy logic` to each subclass

Comment (fsilent) [reply fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/PrivEntry.java]: changed, thank you for your review.

**Comment by fsilent** [comment]
build all

**Comment by fsilent** [comment]
run buildall

**Review by zddr — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**